src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/RegistrarManager.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using Orleans.GrainDirectory;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    /// <summary>
+    /// Maps multi-cluster registration strategies to the corresponding registrar
+    /// </summary>
+    internal class RegistrarManager
```

Comment:
I have checked in a solution where the registrar offers both sync/async versions.

Yes, we could delay this change to the next PR, but I would prefer not. It is easier to remove this again in the next PR if we don't want it  than to remove it now and then re-add it (because of all the dependent branches).